CONTRIBUTING.md

```diff
@@ -0,0 +1,216 @@
+
+```bash
+ ~$ git add some_file.py               # add all the changes you made to this file
+ ~$ git add some_new_file.py           # add a completely new file in its entirety
+ ~$ git add -p docs/some_other_file.py # enter interactive staging mode, to add only portions of a file.
+```
+
+Once you’ve collected all the related changes together on the stage, the `git status` command will now refer to them as _“changes staged for commit”_. You can commit them to the current branch with the [commit](https://git-scm.com/docs/git-commit) command. If you just type `git commit` by itself, git will open the text editor you configured it to use so that you can write a _commit message_ — a short description of the changes you’ve grouped together in this commit. You can bypass the text editor by passing a commit message on the command line with the `-m` flag. For example, if your first commit adds a new feature, your commit message might be:
+
+```bash
+ ~$ git commit -m 'Author: adds feature X'
+```
+
+Once you’ve made the commit, the stage is now empty, and you can repeat the cycle. When you’re done and everything looks good, it’s time to push your changes to your fork:
+
+```bash
+ # push local changes to remote branch origin/new-branch
+ # (this will create the remote branch if it doesn't already exist)
+ ~$ git push origin your-branch
+```
+
+Finally, go to the [Neurosin-lab/netpyne GitHub page](https://github.com/Neurosim-lab/netpyne), click on the pull requests tab, click the “new pull request” button, and choose “compare across forks” to select `your-branch` as the “head repository” and select `development` branch as "base repository".
+
+> **Note:** See the GitHub help page on [creating a PR from a fork](https://help.github.com/en/articles/creating-a-pull-request-from-a-fork) for more information about opening pull requests. To learn more about git, check out the [GitHub help website](https://help.github.com/en), the [GitHub Learning Lab](https://lab.github.com/) tutorial series, and the [pro git book](https://git-scm.com/book/en/v2).
+
+### Connecting to GitHub with SSH (optional)
+
+One easy way to speed up development is to reduce the number of times you have to type your password. SSH (secure shell) allows authentication with pre-shared key pairs. The private half of your key pair is kept secret on your computer, while the public half of your key pair is added to your GitHub account; when you connect to GitHub from your computer, the local git client checks the remote (public) key against your local (private) key, and grants access your account only if the keys fit. GitHub has several help pages that guide you through the process.
+
+Once you have set up GitHub to use SSH authentication, you should change the addresses of your Netpyne-lab GitHub remotes, from `https://` addresses to `git@` addresses, so that git knows to connect via SSH instead of HTTPS.
+>**Note:** You only could have a _SSH key_ from a repository you are a contributor.

```

Comment:
You can only have an _SSH key_ for repositories where you are a contributor.

Reply:
:+1: I change for:
>**Note:** You only have a _SSH key_ for repositories where you are a contributor.